ipn/state_enum_encoding.go

```diff
@@ -0,0 +1,56 @@
+// Code generated by go-enum-encoding DO NOT EDIT
+package ipn
+
+import "errors"
+
+var ErrUnknownState = errors.New("unknown State")
+
+func (s *State) UnmarshalText(text []byte) error {
+	switch string(text) {
+	case "NoState":
+		*s = NoState
+	case "InUseOtherUser":
+		*s = InUseOtherUser
+	case "NeedsLogin":
+		*s = NeedsLogin
+	case "NeedsMachineAuth":
+		*s = NeedsMachineAuth
+	case "Stopped":
+		*s = Stopped
+	case "Starting":
+		*s = Starting
+	case "Running":
+		*s = Running
+	default:
+		return ErrUnknownState
+	}
+	return nil
+}
+
+var json_bytes_State = [...][]byte{[]byte("NoState"), []byte("InUseOtherUser"), []byte("NeedsLogin"), []byte("NeedsMachineAuth"), []byte("Stopped"), []byte("Starting"), []byte("Running")}
+
+func (s State) MarshalText() ([]byte, error) {
+	switch s {
+	case NoState:
+		return json_bytes_State[0], nil
+	case InUseOtherUser:
+		return json_bytes_State[1], nil
+	case NeedsLogin:
+		return json_bytes_State[2], nil
+	case NeedsMachineAuth:
+		return json_bytes_State[3], nil
+	case Stopped:
+		return json_bytes_State[4], nil
+	case Starting:
+		return json_bytes_State[5], nil
+	case Running:
+		return json_bytes_State[6], nil
+	default:
+		return nil, ErrUnknownState
+	}
+}
+
+func (s State) String() string {
+	b, _ := s.MarshalText()
+	return string(b)
+}
```

Comment:
This allocates unnecessarily.

Reply:
true. let me make version without malloc.

Reply:
fixed.

```bash
$ go test -benchmem -run=^$ -bench ^BenchmarkState_String$ tailscale.com/ipn
goos: darwin
goarch: arm64
pkg: tailscale.com/ipn
cpu: Apple M3 Max
BenchmarkState_String-16    	128010190	         9.145 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	tailscale.com/ipn	2.716s
```